include/seqan3/io/detail/misc.hpp

```diff
@@ -76,9 +76,9 @@ void set_format(format_variant_type & format,
     if (extension.size() > 1)
     {
         extension = extension.substr(1); // drop leading "."
-        meta::for_each(valid_formats{}, [&] (auto && fmt)
+        detail::for_each<valid_formats>([&] (auto fmt)
         {
-            using fmt_type = remove_cvref_t<decltype(fmt)>;
+            using fmt_type = remove_cvref_t<typename decltype(fmt)::type>;
```

Comment:
Can you elaborate a bit on the changes going on here? (Why no `&&`, why `typename [...]::type`)